code/game/machinery/computer/dropship_weapons.dm

```diff
@@ -438,189 +673,464 @@
+		to_chat(human_operator, SPAN_WARNING("Fire Mission already underway."))
+		return FALSE
+	editing_firemission = firemission_envelope.missions[firemission_ref]
+	return TRUE
+
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_exit_firemission_edit()
+	editing_firemission = null
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_switch_to_firemission()
+	var/mob/weapons_operator = usr
+	if(!skillcheck(weapons_operator, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
+		to_chat(weapons_operator, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
+		return FALSE
+	in_firemission_mode = TRUE
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_leave_firemission_execution()
+	var/mob/weapons_operator = usr
+	if(!skillcheck(weapons_operator, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
+		to_chat(weapons_operator, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
+		return FALSE
+	firemission_envelope.remove_user_from_tracking(usr)
+	in_firemission_mode = FALSE
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_change_direction()
+	var/mob/M = usr
+	if(!skillcheck(M, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
+		to_chat(usr, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
+		return FALSE
+	var/list/directions = list(dir2text(NORTH), dir2text(SOUTH), dir2text(EAST), dir2text(WEST))
+	var/chosen = tgui_input_list(usr, "Select new Direction for the strafing run", "Select Direction", directions)
+
+	var/chosen_dir = text2dir(chosen)
+	if(!chosen_dir)
+		to_chat(usr, SPAN_WARNING("Error with direction detected."))
+		return FALSE
+
+	update_direction(chosen_dir)
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_change_offset()
+	var/mob/weapons_operator = usr
+	if(!skillcheck(weapons_operator, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
+		to_chat(weapons_operator, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
+		return FALSE
+
+	var/chosen = stripped_input(weapons_operator, "Select Fire Mission length, from 0 to [firemission_envelope.max_offset]", "Select Offset", "[firemission_envelope.recorded_offset]", 2)
+	var/chosen_offset = text2num(chosen)
+
+	if(chosen_offset == null)
+		to_chat(weapons_operator, SPAN_WARNING("Error with offset detected."))
+		return FALSE
+
+	update_offset(chosen_offset)
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/tgui_change_offset(mob/weapons_operator, fm_tag, weapon_id, offset_id, offset_value)
```

Comment:
consistency - other procs are named ui_  - is this normal?

Reply:
A lot of the ui_ procs need removed